data_diff/__init__.py

```diff
@@ -1,9 +1,86 @@
-from typing import Tuple
+from typing import Tuple, Iterator
 
 from .database import connect_to_uri
-from .diff_tables import TableSegment, TableDiffer
+from .diff_tables import (
+    TableSegment,
+    TableDiffer,
+    DEFAULT_BISECTION_THRESHOLD,
+    DEFAULT_BISECTION_FACTOR,
+    DbKey,
+    DbTime,
+    parse_table_name,
+)
 
 
-def create_source(db_uri: str, table_name: str, key_column: str, extra_columns: Tuple[str, ...] = ()):
-    db = connect_to_uri(db_uri)
-    return TableSegment(db, (table_name,), key_column, tuple(extra_columns))
+class TableRef:
+    """Connects to a URI and creates a table ref
+
+    This is an auxiliary class for diff_tables()
+    """
+
+    def __init__(self, db_uri: str, table_name: str):
+        self.db = connect_to_uri(db_uri)
+        self.table_name = table_name
+
+    def create_table_segment(self, **kwargs):
+        return TableSegment(self.db, parse_table_name(self.table_name), **kwargs)
+
+
+def diff_tables(
+    table1: TableRef,
+    table2: TableRef,
+    *,
+    # Name of the key column, which uniquely identifies each row (usually id)
+    key_column: str = "id",
+    # Name of updated column, which signals that rows changed (usually updated_at or last_update)
+    update_column: str = None,
+    # Extra columns to compare
+    extra_columns: Tuple[str, ...] = (),
+    # Start/end key_column values, used to restrict the segment
+    start_key: DbKey = None,
+    end_key: DbKey = None,
+    # Start/end update_column values, used to restrict the segment
+    min_time: DbTime = None,
+    max_time: DbTime = None,
+    # Into how many segments to bisect per iteration
+    bisection_factor: int = DEFAULT_BISECTION_FACTOR,
+    # When should we stop bisecting and compare locally (in row count)
+    bisection_threshold: int = DEFAULT_BISECTION_THRESHOLD,
+    # Enable/disable threaded diffing. Needed to take advantage of database threads.
+    threaded: bool = True,
+    # Maximum size of each threadpool. None = auto. Only relevant when threaded is True.
+    # There may be many pools, so number of actual threads can be a lot higher.
+    max_threadpool_size: int = None,
```

Comment:
Why isn't this exactly symmetric with the CLI options?

Reply:
This way gives the programmer more control. The CLI is a simplification, to make it easier.

Or if you mean why the default isn't 1, I just missed it and will fix.